mono/metadata/object.c

```diff
@@ -4510,12 +4510,14 @@ mono_object_new (MonoDomain *domain, MonoClass *klass)
  * For SGEN, these objects will only be freed at appdomain unload.
  */
 MonoObject *
-mono_object_new_pinned (MonoDomain *domain, MonoClass *klass)
+mono_object_new_pinned (MonoDomain *domain, MonoClass *klass, MonoError *error)
 {
 	MONO_REQ_GC_UNSAFE_MODE;
 
 	MonoVTable *vtable;
 
+	mono_error_init (error);
+
 	vtable = mono_class_vtable (domain, klass);
 	if (!vtable)
 		return NULL;
```

Comment:
The existing behavior is wrong and we should not keep it that way. As we increase
our reliance on MonoError, this sort of problem will only get worse.

We should fix it and leave a FIXME to further weave mono error as we go.